gateway/api/use_cases/jobs/provider_logs.py

```diff
@@ -55,10 +56,21 @@ def execute(self, job_id: UUID, user: AbstractUser) -> str:
 
         logs = logs_storage.get(job_id)
 
-        if logs is None:
-            logs = job.logs
+        # No logs stored.
+        if logs:
+            return logs
+
+        # Get from Ray if it is already running.
+        if job.compute_resource:
```

Comment:
I would check for the compute_resource before COS. So, the order will be: 1) computer resource? ray. 2) COS? cos 3) db